nion/swift/ExportDialog.py

```diff
@@ -34,167 +36,191 @@
 _ = gettext.gettext
 
 
-class ExportDialog(Dialog.OkCancelDialog):
-    def __init__(self, ui: UserInterface.UserInterface, parent_window: Window.Window):
-        super().__init__(ui, ok_title=_("Export"), parent_window=parent_window)
+class ExportDialogViewModel:
+    include_title: Model.PropertyModel[bool]
+    include_date: Model.PropertyModel[bool]
+    include_dimensions: Model.PropertyModel[bool]
+    include_sequence: Model.PropertyModel[bool]
+    include_prefix: Model.PropertyModel[bool]
+    prefix: Model.PropertyModel[str]
+    directory: Model.PropertyModel[str]
+    writer: Model.PropertyModel[ImportExportManager.ImportExportHandler]
```

Comment:
These are not necessary to declare here since they are initialized in the `__init__` method. I'm guessing you were messing around with `dataclass` which requires you to put them here.

Reply:
I can't remember if this was one I did when battling with the type-checker on pull or not, but is there a downside to doing this (aside from more words). It seems throughout the rest you are trying to be as type specific as possible, so trying to find where that line is.

Reply:
I'll refer you to some sections of Python (first link is general section about classes; 2nd link is specific answer to your question above):

https://docs.python.org/3/tutorial/classes.html#class-objects
https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables

In a nutshell, instance variables should be declared in `__init__` and in doing so, they will have enough typing information to fully specify the type. So no need for the redundant typing information, i.e. line 40 is redundant due to line 50 (the correct line).